FreeRTOS/Source/tasks.c

```diff
@@ -4128,6 +4138,12 @@ TCB_t *pxTCB;
 					pxMutexHolderTCB->uxPriority = pxCurrentTCB->uxPriority;
 				}
 
+#if ( configMUTEX_MULTI_STEP_PRIORITY_DISINHERITANCE == 1 )
+				listSET_LIST_ITEM_VALUE(xListItem, ( ( TickType_t ) configMAX_PRIORITIES - ( TickType_t ) pxMutexHolderTCB->uxPriority ));
+				uxListRemove( xListItem );

```

Comment:
@avtolstoy why is the remove after setting value?

Reply:
We insert it right back in immediately after, so that it's placed in the appropriate location priority-wise (see `vListInsert()` implementation).